src/basics.md

```diff
@@ -1101,14 +1101,109 @@ fn main() {
 }
 ```
 
+[ex-error-chain-backtrace]: #ex-error-chain-backtrace
+<a name="ex-error-chain-backtrace"></a>
+## Obtain backtrace of complex error scenarios
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+This recipe shows how to handle a complex error scenario and then
+print a backtrace. It relies on [`chain_err`] to extend errors by
+appending new errors. The error stack can be unwinded, thus providing
+a better context to understand why an error was raised.
+
+The below recipes attemps to deserialize the value `256` into a
+`u8`. An error will bubble up from Serde then csv and finaly up to the
+user code.
+
+```rust
+# extern crate csv;
+#[macro_use]
+extern crate error_chain;
+# #[macro_use]
+# extern crate serde_derive;
+#
+# use std::fmt;
+#
+# error_chain! {
+#     foreign_links {
+#         Reader(csv::Error);
+#     }
+# }
+
+#[derive(Debug, Deserialize)]
+struct Rgb {
+    red: u8,
+    blue: u8,
+    green: u8,
+}
+
+fn read_rgb_from_csv(csv_data: &[u8]) -> Result<Rgb> {
+    let color: Rgb = csv::Reader::from_reader(csv_data)
+        .deserialize()
+        .nth(0)
+        .ok_or("Cannot deserialize the first CSV record")?
+        .chain_err(|| "Cannot deserialize RGB color")?;
+
+    Ok(color)
+}
+
+# impl fmt::UpperHex for Rgb {
+#     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+#         let hexa = u32::from(self.red) << 16 | u32::from(self.blue) << 8 | u32::from(self.green);
+#         write!(f, "{:X}", hexa)
+#     }
+# }
+#
+fn run() -> Result<()> {
+    let csv = "red,blue,green
+102,256,204";
+
+    let rgb = read_rgb_from_csv(csv.as_bytes()).chain_err(|| "Cannot read CSV data")?;
+    println!("{:?} to hexadecimal #{:X}", rgb, rgb);
+
+    Ok(())
+}
+
+fn main() {
+    if let Err(ref errors) = run() {
+        eprintln!("Error level - description");
+        errors
+            .iter()
+            .enumerate()
+            .for_each(|(index, error)| eprintln!("└> {} - {}", index, error));
+
+        if let Some(backtrace) = errors.backtrace() {
+            eprintln!("{:?}", backtrace);
+        }
+
+        ::std::process::exit(1);
```

Comment:
I'd just drop this line to make our CI happy or make the example `no_run`. We can always write a comment here mentioning that we would normally handle the errors or leave the app here with and error.

Reply:
The CI is trying to execute the recipes by default, meaning it checks the code compiles at least. If `no_run` is set, is the code still compiled?

I try to understand which is better `no_run` vs make the code return a success code.

Reply:
`no_run` - compiles the code but does not run it
`ignore` - does not compile it at all

I'd probably go with leaving this example runnable but removing the `::std::process::exit(1);` but I'll leave it up to you.

Reply:
Thanks for the listing.

> I'd probably go with leaving this example runnable but removing the ::std::process::exit(1);

Yes, I'll do that